src/terralib.lua

```diff
@@ -1651,28 +1651,38 @@ do
     --map from luajit ffi ctype objects to corresponding terra type
     types.ctypetoterra = {}
     
-    local function globaltype(name, typ)
+    local function globaltype(name, typ, min_v, max_v)
         typ.name = typ.name or name
         rawset(_G,name,typ)
         types[name] = typ
+        if min_v then function typ:min() return terra.cast(self, min_v) end end
+        if max_v then function typ:max() return terra.cast(self, max_v) end end
     end
     
     --initialize integral types
     local integer_sizes = {1,2,4,8}
     for _,size in ipairs(integer_sizes) do
         for _,s in ipairs{true,false} do
-            local name = "int"..tostring(size * 8)
+            local bits = size * 8
+            local name = "int"..tostring(bits)
             if not s then
                 name = "u"..name
             end
+            local min,max
+            if not s then
+                min = 0ULL
+                max = 2ULL ^ (bits - 1)
```

Comment:
Shouldn't this be `(2ULL ^ bits) - 1`? Technically this does wrap about at 64 bits, but that seems to work on LuaJIT 2.0.5.

Reply:
Yup, that was incorrect. Commit 0efbb39 should address this.